misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/Formatter.java

```diff
@@ -94,19 +96,20 @@ public static SyntaxTree format(SyntaxTree syntaxTree, LineRange range, Formatti
      * @param syntaxTree The SyntaxTree which is to be formatted
      * @param options Formatting options that are to be used when formatting
      * @return The modified SyntaxTree after formatting changes
+     * @throws FormatterException Exception caught while formatting
      */
-    public static SyntaxTree format(SyntaxTree syntaxTree, FormattingOptions options) {
+    public static SyntaxTree format(SyntaxTree syntaxTree, FormattingOptions options) throws FormatterException {
         return modifyTree(syntaxTree, options, null);
     }
 
-    private static SyntaxTree modifyTree(SyntaxTree syntaxTree, FormattingOptions options, LineRange range) {
+    private static SyntaxTree modifyTree(SyntaxTree syntaxTree, FormattingOptions options, LineRange range)
+            throws FormatterException {
         FormattingTreeModifier treeModifier = new FormattingTreeModifier(options, range);
         ModulePartNode modulePartNode = syntaxTree.rootNode();
         try {
             return syntaxTree.modifyWith(treeModifier.transform(modulePartNode));
         } catch (Exception e) {
-            LOGGER.error(String.format("Error while formatting the source: %s", e.getMessage()));
-            return syntaxTree;
+            throw new FormatterException("Error while formatting the source: " + e.getMessage(), e.getCause());
```

Comment:
Does `e.getMessage()` always return the source name? Otherwise let's change the exception message

Reply:
Fixed